### PR TITLE
fixed the time attack bug

### PR DESCRIPTION
**Pull request by filipnyquist**
Fixed the time attack bug in issue #405

### CODE REVIEW

**Comment by filipnyquist** [comment]
Needs to be tested by someone else aswell.

File: app/Http/Middleware/HMACAuthorization.php [hunk 1]
```diff
@@ -170,7 +170,7 @@ protected function validateIPAccess()
      */
     protected function validateContents()
     {
-        if (base64_decode($this->hash()) !== $this->generateSignature()) {
+        if(!hash_equals(base64_decode($this->hash()),$this->generateSignature())){
```
Comment (DaneEveritt) [inline app/Http/Middleware/HMACAuthorization.php]: As a general code standard, we have a space between `if` and the parenthesis, `!` and functions, as well as after commas.

```
if (! hash_equals(..., ...))
```

**Comment by DaneEveritt** [comment]
Aside from the code formatting, LGTM 👍

**Comment by DaneEveritt** [comment]
Thanks for the PR, glad you found the formatting there haha. :)

**Pull request merged**